File: CS/CR_CodeDecorationBlueBox/CR_CodeDecorationBlueBox/BlueAdornmentFactory.cs
using System;
using System.Collections.Generic;

using DevExpress.DXCore.Adornments;
using DevExpress.DXCore.Platform.Drawing;

using DevExpress.CodeRush.Core;
using DevExpress.CodeRush.StructuralParser;

namespace CR_CodeDecorationBlueBox
{
    class BlueBoxAdornmentFactory : TextDocumentAdornment
    {
        public BlueBoxAdornmentFactory(SourceRange range)
            : base(range)
        {
        }

        protected override TextViewAdornment NewAdornment(string feature, IElementFrame frame)
        {
            return new BlueBoxAdornment(feature, frame);
        }
    }
}

File: VB/CR_CodeDecorationBlueBox/CR_CodeDecorationBlueBox/BlueAdornmentFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.DXCore.Adornments
Imports DevExpress.CodeRush.StructuralParser

' This class is instanciated by the DXCore per TextDocument.
' It's NewAdornment function is called whenever the DXCore needs to create an adornement
Class BlueBoxAdornmentFactory
    Inherits TextDocumentAdornment
    Public Sub New(ByVal range As SourceRange)
        MyBase.New(range)
    End Sub
    Protected Overrides Function NewAdornment(ByVal feature As String, ByVal frame As IElementFrame) As TextViewAdornment
        Return New BlueBoxAdornment(feature, frame)
    End Function
End Class

